test/hooks-async.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const request = require('request')
+const sleep = require('then-sleep')
+const fastify = require('..')()
+
+function asyncHookTest (t) {
+  const test = t.test
+  test('hooks - add async preHandler', t => {
+    t.plan(1)
+    try {
+      fastify.addHook('preHandler', async function (request, reply, next) {
```

Comment:
`next`  should not be tampered with if we are returning a promise. They are mutually exclusive, however we don't have any _fast_ means to check this. Let's just assume it's not passed in, ok?

Reply:
Yes, `next` should not be used inside the async handler.

Reply:
can you remove them from here as well?

Reply:
It should be.